utils/errors.go

```diff
@@ -0,0 +1,44 @@
+/*
+This file is part of REANA.
+Copyright (C) 2022 CERN.
+
+REANA is free software; you can redistribute it and/or modify it
+under the terms of the MIT License; see LICENSE file for more details.
+*/
+
+package utils
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"net/url"
+	"reflect"
+)
+
+var UnexpectedErrorMsg = "unexpected error"
+
+// HandleBasicApiError Handles API Error response which contains a payload with a message
+// Returns an unexpected error when this doesn't happen
+func HandleBasicApiError(err error) error {
+	urlErr, isUrlErr := err.(*url.Error)
+	if isUrlErr {
+		return fmt.Errorf(
+			"'%s' not found, please verify the provided server URL or check your internet connection",
+			urlErr.Err.(*net.OpError).Err.(*net.DNSError).Name,
+		)
+	}
+
+	apiErr := reflect.ValueOf(err).Elem()
+	apiErrType := apiErr.Type()
+	if apiErr.NumField() > 0 && apiErrType.Field(0).Name == "Payload" {
+		payload := apiErr.Field(0).Elem()
+		payloadType := payload.Type()
+		if payload.NumField() > 0 && payloadType.Field(0).Name == "Message" {
+			msg := payload.Field(0).String()
+			return errors.New(msg)
+		}
+	}
+
+	return errors.New(UnexpectedErrorMsg)
```

Comment:
The idea was to return an understandable error and, at the same time, have a way of detecting that the error wasn't handled (comparing the constant) so it's possible for a command to handle it if needed. However, I may see a better way to do this:
We could return the original error and a boolean informing if the error was handled or not. What do you think?

Reply:
This would not yet be used so I'm not sure if it's worth it doing right now (YAGNI principle I guess)

Reply:
I agree that it's probably not useful to know if the error was handled or not right now.

In any case, if we return the original error, wouldn't it be possible to use this function on any type of error? If so, we could move it to `PostRun` as suggested here: https://github.com/reanahub/reana-client-go/pull/55#discussion_r935461887. By doing so, we wouldn't need to call `HandleBasicApiError` each time we use the API, right? What do you think about this?

Reply:
I agree, that would be the best case scenario, as long as we're able to do the openapi changes

Reply:
@audrium @mdonadoni On further inspection, we don't even need to change the openapi, since the handleBasicApiError doesn't panic anymore, it just returns any different error